src/realm/object-store/dictionary.hpp

```diff
@@ -28,21 +28,53 @@ namespace object_store {
 
 class Dictionary : public object_store::Collection {
 public:
+    using Iterator = realm::Dictionary::Iterator;
     Dictionary() noexcept;
     Dictionary(std::shared_ptr<Realm> r, const Obj& parent_obj, ColKey col);
     Dictionary(std::shared_ptr<Realm> r, const realm::Dictionary& list);
     ~Dictionary() override;
 
+    bool operator==(const Dictionary& rgt) const noexcept
+    {
+        return *m_dict == *rgt.m_dict;
+    }
+
     template <typename T>
     void insert(StringData key, T value);
     template <typename T>
     T get(StringData key) const;
 
+    void erase(StringData key)
+    {
+        verify_in_transaction();
+        m_dict->erase(key);
+    }
+    void remove_all()
+    {
+        verify_in_transaction();
+        m_dict->clear();
+    }
 
     template <typename T, typename U, typename Context>
     void insert(Context&, T&& key, U&& value, CreatePolicy = CreatePolicy::SetLink);
     template <typename Context, typename T>
     auto get(Context&, T key) const;
+    template <typename Context, typename T>
+    void erase(Context&, T key) const;
```

Comment:
I don't think we need a Context version of this or for any of the functions to have the key type be a template. Since it's always a string doing the unboxing in the caller is trivial.

Reply:
Alright. That just makes it simpler.